docs/testing.md

```diff
@@ -0,0 +1,130 @@
+
+They can be handy as API snapshot tests. For example, to ensure that a `serde` implementation doesn't change.
+
+#### Integration tests
+
+These are tests that depend on multiple components (perhaps across module boundaries).
+Additionally, this kind of tests usually involve mocked state.
+
+We are aiming to not spend too much time maintaining Rust integration tests, in favour of more comprehensive tests.
+
+### The `app` frontend
+
+This includes Flutter code (written in Dart) and the generated `flutter_rust_bridge`.
+In practice, there shouldn't be a strong need to write tests for the `flutter_rust_bridge` code we write (Rust) since it wil be a thin layer.
+
+#### Unit tests
+
+We can apply similar principles as the ones mentioned for Rust code.
+
+#### Widget tests
+
+Used to verify Flutter widget appearance and behaviour.
+They target a single widget.
+The state is mocked i.e. no Rust called via `flutter_rust_bridge`.
+
+#### Integration tests
+
+These are your traditional "UI tests".
+The clicks on the UI are simulated as if someone was using the application.
+
+They remain lightweight as they only call to a mocked version of the API provided by `flutter_rust_bridge`.
+We do not currently specify how this API shoud be mocked.
+We considered (1) [mocking the dependency in Dart](https://docs.flutter.dev/cookbook/testing/unit/mocking) and (2) mocking the dependency in Rust.
+We would prefer (1) as we think it would lead to simpler code than (2), but we are not sure if it's a feasible approach.
+
+### System tests
+
+These tests verify behaviour when multiple systems run in tandem.
+When enough systems are involved, these might be referred to as end-to-end (e2e) tests.
+
+#### Combinations
+
+Here we list all the possible useful combinations we can imagine:
+
+1. `App` backend + `coordinator`.
+2. `App` backend + blockchain.
+3. `Coordinator` + blockchain.
+4. `App` backend + `coordinator` + blockchain.
+
+Notably, the `app` frontend is excluded from system tests.
+This is feasible if we ensure that the `app` frontend doesn't control important state for the `app`.
+
+Thusly, we recommend that system tests involving the `app` backend be written in Dart.
+These tests will be run using the [`dart test` command](https://dart.dev/tools/dart-test).
+These tests will target the `flutter_rust_bridge` Dart API generated by the `app` backend.
+
+The details of how the test environment should bet set up or consumed are not yet specified.
+Regardless, we should aim to be able to conveniently run these tests locally by providing `docker-compose` files and/or setup scripts or by connecting to an always-available semi-public environment.
+
+This document does not specify what type of Bitcoin blockchain these tests should run against.
+We do believe it would be important to be able to run them against `signet` and `testnet`.
+
+### Load tests
+
+We should leverage a semi-public environment to assess the scalability of our solution.
```

Comment:
Which component/part do you want to assess exactly here?

Reply:
This section is not fleshed out, but we were referring to the whole system (`app` + `composer` + `blockchain`).

Daniel pointed out that our testnet setup was actually just load testing.

> Which component/part do you want to assess exactly here?

The question still stands though. I think it would make sense to eventually load test the `composer`, to figure out how many users it can serve. Like we used to do in ItchySats. In ItchySats we never quite solved all the problems related to load, so it would be good to avoid that situation in 10101.

Reply:
whats the `composer`? do you mean the `coordinator`?